core-java/src/main/java/com/baeldung/bufferedreader/BufferedReaderExample.java

```diff
@@ -0,0 +1,108 @@
+package com.baeldung.bufferedreader;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Arrays;
+
+public class BufferedReaderExample {
+
+    public String readAllLines(BufferedReader reader) throws IOException {
+        StringBuilder content = new StringBuilder();
+        String line;
+
+        while ((line = reader.readLine()) != null) {
+            content.append(line);
+            content.append(System.lineSeparator());
+        }
+
+        return content.toString();
+    }
+
+    public String readAllLines2(BufferedReader reader) {
```

Comment:
Instead of `readAllLines2` could we use the method name to describe the different approach. e.g. `readAllLinesWithStreams`.